### DISCUSSION TITLE
i18n Support

### DISCUSSION THREAD

**Discussion by Insecty** [ggarra13/mrv2#292] [Ideas]
Hi, I wonder will you support Chinese or other languages?

and i wonder if i want to add chinese language support, is there some documents or tutorials that i can refer to?
Thanks!

**Comment by ggarra13** [comment]
> Hi, I wonder will you support Chinese or other languages?

Certainly!  Currently Spanish and English are supported, as those are the languages I speak.  But the framework is setup to support any language.  See Window->Preferences->Language and Colors.

> and i wonder if i want to add chinese language support, is there some documents or tutorials that i can refer to?

Translations are based on gettext and libintl.  This involves adding and modifying .po files.  You may need to clone the mrv2 repository and create a zh.po (is that for Chinese? I forget) file.  For testing you should become familiar with compiling mrv2.

See the sample mrv2/po/es.po.  For more info on the workflow, see:

https://github.com/ggarra13/mrv2#translating

Feel free to ask if you get stuck or need some help!

**Comment by ggarra13** [comment]
I've updated the framework for Simplified Chinese and Traditional Chinese in the github repository and the mrv2 interface.

Just clone the mrv2 repository, look for mrv2/po/zh-CN.po and translate the messages.

If you want to test the messages in the UI, you will need to convert the .po files into .mo files.  Check the translating instructions I mentioned above.

**Comment by ggarra13** [comment]
I have done automatic translations of Chinese (Simplified) and German with AI in v1.2.5.  You can play with it in:

https://sourceforge.net/projects/mrv2/files/beta/

**Reply by Insecty** [reply]
Hi, there is no .mo file in compiled folder.
![image](https://github.com/user-attachments/assets/7938547a-a6a2-433d-9c13-6e772ec332fa)
Thus when i click the language setting in preference, there is no change~

**Comment by ggarra13** [comment]
mrv2 v1.2.5 is now out.  It has Chinese, German and Hindi translations done with AI.  Spanish and English are manual translations.
If you want to correct errors in the AI translations, do this:

1) Fork a copy of mrv2's repository.
2) Checkout a local copy of your own mrv2 repository with git, like:

         git clone git@github.com:insecty/mrv2.git

3) With a good text editor, edit:

         mrv2/po/zh-CN.po
         
msgid is the original message while msgstr is the translated message.

4) Commit your changes.

        git commit -a
                
5) Create a pull request  on Github from your main repository to ggarra13/mrv2's main repository.

6) Add some comments saying what you corrected.

**Reply by Insecty** [reply]
i will have a try, thank you so much!!!

**Reply by Insecty** [reply]
I have reviewed the translation from AI, and commit a pull request~

**Reply by ggarra13** [reply]
Thanks.  I merged your pull request (I had to do two minor technical fixes to it).  I also fixed the automated beta build scripts to create the .mo files.  I am compiling a new version of v1.2.6 with your translations now.

**Comment by ggarra13** [comment]
I am also attaching a simple gettext tutorial which lists the gettext commands you can run manually to update the .mo files in case you want to do further refinements on mrv2's UI (like check margins, python help docs, etc):

https://www.labri.fr/perso/fleury/posts/programming/a-quick-gettext-tutorial.html
